common/ActionStatusSummary.ts

```diff
@@ -36,27 +36,38 @@ const getCacheKey = (
 
 export const getStatusSummary = memoize(_getStatusSummary, getCacheKey);
 
-export const getStatusColor = (color: string, theme: Theme) => {
+export const getThemeColor = (color: string, theme: Theme) => {
   return theme.graphColors[color];
 };
 
-interface ActionWithStatusSummary {
-  statusSummary?: { color?: string; identifier?: string };
+export interface ActionWithStatusSummary {
+  statusSummary?: {
+    color?: string;
+    identifier?: ActionStatusSummaryIdentifier;
+    label?: string;
+  };
+  color?: string | null;
 }
 
 export const getStatusColorForAction = (
   action: ActionWithStatusSummary,
   plan: PlanContextType,
   theme: Theme
 ) => {
-  const c = action?.color;
-  if (c != null) {
-    return getStatusColor(c, theme);
+  const { color } = action;
+  if (color != null) {
+    return getThemeColor(color, theme);
   }
-  const s = action?.statusSummary;
-  if (s == null || (s?.identifier == null && s?.color == null)) {
+  const { statusSummary } = action;
+  if (statusSummary == null) {
```

Comment:
No need to change this since it was pre-existing, but in what scenario would `statusSummary` be null/ undefined? Throwing errors without rigid try / catch blocks in place gives me the heebie-jeebies. Here we could fall back to a default colour and log a Sentry error instead.

Reply:
Yeah, it would be a programming error, some obscure graphql query where we would have forgotten to add the field.You're right, Sentry was the only point to have this, but we haven't received any errors. I'll just skip the throwing altogether.